lib/ruby_indexer/test/index_test.rb

```diff
@@ -1472,5 +1472,26 @@ class Foo
 
       assert_empty(@index.method_completion_candidates("bar", "Foo"))
     end
+
+    def test_resolving_constants_in_aliased_namespace
+      index(<<~RUBY)
+        module Original
+          module Something
+            CONST = 123
+          end
+        end
+
+        module Other
+          ALIAS = Original::Something
+        end
+
+        module Third
+          Other::ALIAS::CONST
+        end
+      RUBY
+
+      entry = T.must(@index.resolve("Other::ALIAS::CONST", ["Third"])&.first)
```

Comment:
Let's also add a test for top levels constant aliases, I don't think we have one already:

```rb
class Foo
  CONST = 123
end

FOO = Foo
FOO::CONST
```